beetmoverscript/task.py

```diff
@@ -152,3 +157,25 @@ def update_props(context, props, platform_mapping):
     props["platform"] = platform_mapping.get(stage_platform, stage_platform)
     props["stage_platform"] = stage_platform
     return props
+
+
+def get_updated_buildhub_artifact(path, installer_path, context, manifest, locale):
+    """
+    Read the file into a dict, alter the fields below, and return the updated dict
+    buildhub.json fields that should be changed: download.size, download.date, download.url
+    """
+    contents = utils.load_json(path)
+    installer_name = os.path.basename(installer_path)
+    dest = manifest['mapping'][locale][installer_name]['destinations']
+    url_prefix = context.config["bucket_config"][context.bucket]["url_prefix"]
+
+    # Update fields
+    contents['download']['size'] = utils.get_size(installer_path)
+    contents['download']['date'] = str(arrow.utcnow())
+    
+    url = os.path.join(manifest["s3_bucket_path"], dest[0])
+    # URL encode the url; we are assuming the url prefix is ascii safe
+    url = '/'.join([urllib.parse.quote(x) for x in url.split('/')])
+    contents['download']['url'] = url_prefix + "/" + url
```

Comment:
I'd rather join these two strings with the help of a function, rather than this. I'm always feeling unsafe when doing these operations by hand using string concatenation. We can maybe use [urllib.parse.urljoin](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin) or simply the `os.path.join` from line 176 or something.

Reply:
+1 on `urllib.parse.urljoin` 
Equally, no need to use `os.path.join` or `url.split('/')`.

Reply:
got it, hopefully it's good this time 😅